MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -20,12 +20,18 @@ final class FeedViewModel {
 	
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onFeedLoadError: Observer<Error?>?
```

Choosing which message to show is a presentation decision - it should be done by the view model.

`onFeedLoadError: Observer<Error>?` could be `onErrorStateChange: Observer<String?>?`, so the ViewModel decides which error to show and when to show/hide errors (instead of the view controller). It's often easier to manage state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed.

And on failure, the ViewModel can call  `onErrorStateChange(Localized.Feed.loadError)` to show the error.

This way, the ViewController doesn't need to decide which error to show/hide.